### PR TITLE
Storages: Don't merge SegmentReadTasks when data sharing is disabled

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #8564

Problem Summary:

- In PR #8567, we disable data sharing when `dt_max_sharing_column_bytes_for_all <= 0`. 
- However, `SegmentReadTaskScheduler` will still try to merge `SegmentReadTask` when `dt_max_sharing_column_bytes_for_all <= 0`, although these SegmentReadTasks would not sharing data.

### What is changed and how it works?

- Don't merge SegmentReadTasks when data sharing is disabled.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/ReadThread/SegmentReadTaskScheduler.h [hunk 1]
```diff
@@ -86,12 +88,14 @@ class SegmentReadTaskScheduler
...
-    std::atomic<bool> stop;
+    std::atomic<bool> stop{false};
     std::thread sched_thread;
 
     LoggerPtr log;
 
     // To count how many threads are waitting to add tasks.
     std::atomic<Int64> add_waittings{0};
+
+    bool enable_data_sharing{true};
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ReadThread/SegmentReadTaskScheduler.h:99]: Move it next to the `std::atomic<bool> stop` can share the align and reduce memory consumption of this instance. (Though we will only create one instance for this scheduler)

**Review by JaySon-Huang — COMMENTED** [review]
Will it do any harm if we keep merging the SegmentReadTasks when data sharing is disabled?

**Review by JaySon-Huang — APPROVED** [review]
/approve

**Pull request merged**